tex/gregoriotex-syllable.tex

```diff
@@ -648,42 +649,50 @@
   % there are two different cases that have almost nothing in common : the case where there is something written under the bar, and the case where there is nothing.
   % first of all we need to calculate previousenddifference, begindifference, enddifference and nextbegindifference.
-  \gre@calculate@textaligncenter{#1}{#2}{0}%
-  \setbox\GreSyllabletext=\hbox{\grefixedtextformat{#1#2#3}}%
+  \ifnum\gre@englishcentering=1\relax %
+    \gdef\gre@firstsyllablepart{}%
+    \gdef\gre@middlesyllablepart{#1#2#3}%
+    \gdef\gre@endsyllablepart{}%
+  \else %
+    \gdef\gre@firstsyllablepart{#1}%
+    \gdef\gre@middlesyllablepart{#2}%
+    \gdef\gre@endsyllablepart{#3}%
+  \fi %
+  \gre@calculate@textaligncenter{\gre@firstsyllablepart}{\gre@middlesyllablepart}{0}%
+  \setbox\GreSyllabletext=\hbox{\grefixedtextformat{\gre@firstsyllablepart\gre@middlesyllablepart\gre@endsyllablepart}}%
+  \gre@debug{Width of bar text: \the\wd\GreSyllabletext}%
   % we need to save lastoflinecount, because otherwise when a \newline appears inside a syllable, it is counted two times (once here and once later when #9 will be really output), so \grelastoflinecount has a wrong value. Thus
   \xdef\gresavedlastoflinecount{\number\grelastoflinecount\relax}%
   % a counter to know we are not really typesetting things (useful for end of lines)
   \xdef\greboxing{1}%
-  \gresyllablenotes{#9}%
+  \gresyllablenotes{#9}% For some reason, the width of this box is not the width of the bar lines.  However, we keep the command because it effects a call to \grewritebar which stores the width of the bar lines in lastglyphwidth as part of its function.  We'll use that in the remainder of this macro.
+  \gre@debug{Width of bar line: \the\gre@dimen@lastglyphwidth}%
   \xdef\greboxing{0}%
   \grefirstglyph=1\relax %
   \global\grelastoflinecount=\gresavedlastoflinecount\relax %
-  \gre@dimen@notesaligncenter=\wd\GreSyllablenotes%
+  \gre@dimen@notesaligncenter=\gre@dimen@lastglyphwidth%
   \divide\gre@dimen@notesaligncenter by 2\relax %
   \gre@dimen@begindifference=\gre@dimen@notesaligncenter %
   \advance\gre@dimen@begindifference by -\gre@dimen@textaligncenter %
-  \gre@calculate@enddifference{\wd\GreSyllablenotes}{\wd\GreSyllabletext}{\gre@dimen@textaligncenter}{\gre@dimen@notesaligncenter}{1}%
+  \gre@calculate@enddifference{\gre@dimen@lastglyphwidth}{\wd\GreSyllabletext}{\gre@dimen@textaligncenter}{\gre@dimen@notesaligncenter}{1}%
   #5%
   \gre@calculate@nextbegindifference{\gre@nextfirstsyllablepart}{\gre@nextmiddlesyllablepart}{\gre@nextendsyllablepart}{#7}%
   \greunsetfixednexttextformat %
-  % for LuaTeX syllable counting in optimize_gabc, we put an empty syllable box with gregorioattr set to 2. With this we can count every syllable.
-  \gregorioattr=2\relax %
```

Comment:
Overlapping thought processes.  You mentioned elsewhere that the `optimize_*` stuff was being removed.  Since these lines are related to that, I removed them.  That removal, though, doesn't strictly belong here, but with the removal of the `optimize_` stuff.